### DISCUSSION TITLE
Inspecting generated kernel code?

### DISCUSSION THREAD

**Discussion by ib00** [NVIDIA/warp#53] [Q&A] [answered]
Is it possible to inspect the code that was generated for user kernels (for both forward and backward pass)?

**Comment by mmacklin** [comment, marked as answer]
Hi there,

Yes, please take a look in the `Kernel Cache` location printed at Warp initialization time, e.g.: 

```Warp initialized:
   Version: 0.5.0
   Devices:
     "cpu"    | AMD64 Family 25 Model 33 Stepping 0, AuthenticAMD
     "cuda:0" | NVIDIA GeForce RTX 3090
     "cuda:1" | NVIDIA GeForce RTX 2080 Ti
   Kernel cache: C:\Users\mmacklin\AppData\Local\NVIDIA Corporation\warp\Cache\0.5.0
```

`C:\Users\mmacklin\AppData\Local\NVIDIA Corporation\warp\Cache\0.5.0` holds the generated C++/CUDA code and compiled .ptx files that you can inspect. You can also compile / link these into your own existing C++ projects.

Cheers,
Miles
